src/generator/templates.cpp

```diff
@@ -339,6 +340,28 @@ const char *Templates::ClientMethodDefinitionQmlTemplate = "\nvoid $classname$::
                                                            "        QJSValue(errorCallback).call(QJSValueList{jsEngine->toScriptValue(status)});\n"
                                                            "    });\n"
                                                            "}\n";
+const char *Templates::ClientMethodDefinitionQml2Template = "\nvoid $classname$::$method_name$($param_type$ *$param_name$, $return_type$ *$return_name$, const QJSValue &errorCallback)\n"
+                                                            "{\n"
+                                                            "    if ($return_name$ == nullptr) {\n"
+                                                            "        qProtoWarning() << \"Invalid argument provided for method $classname$::$method_name$, argument of type '$return_type$ *' expected\";\n"
+                                                            "        return;\n"
+                                                            "    }\n\n"
+                                                            "    if ($param_name$ == nullptr) {\n"
+                                                            "        qProtoWarning() << \"Invalid argument provided for method $classname$::$method_name$, argument of type '$param_type$ *' expected\";\n"
+                                                            "        return;\n"
+                                                            "    }\n\n"
+                                                            "    QJSEngine *jsEngine = qjsEngine(this);\n"
+                                                            "    if (jsEngine == nullptr) {\n"
+                                                            "        qProtoWarning() << \"Unable to call $classname$::$method_name$, it's only callable from JS engine context\";\n"
+                                                            "        return;\n"
+                                                            "    }\n\n"
+                                                            "    QtProtobuf::QGrpcAsyncReplyShared reply = call(\"$method_name$\", *$param_name$);\n"
+                                                            "    reply->subscribe(jsEngine, [this, reply, jsEngine, $return_name$]() {\n"
+                                                            "        *$return_name$ = $return_type$(reply->read<$return_type$>());\n"
```

Comment:
Here we have potential risk that "return" object will be destroyed at point when callback actually called. The easy solution that I see is to use QPointer in the method and copy it to lambda.

QPointer<$return_type$> safeReturn($return_name$);
...
    reply->subscribe(jsEngine, [this, reply, jsEngine, safeReturn]() {//as copy, it's important
    if (safeReturn.isNull()) {
         qProtoWarning() << "Return value is destroyed. Ignore call result";
         return;
    }
    *safeReturn = $return_type$(reply->read<$return_type$>());
...

Reply:
@semlanik Thanks for the catch. I have pushed a fix